kratos/geometries/hexahedra_3d_20.h

```diff
@@ -878,11 +878,16 @@ template<class TPointType> class Hexahedra3D20 : public Geometry<TPointType>
      */
     void PrintData( std::ostream& rOStream ) const override
     {
+        // Base Geometry class PrintData call
         BaseType::PrintData( rOStream );
         std::cout << std::endl;
-        Matrix jacobian;
-        this->Jacobian( jacobian, PointType() );
-        rOStream << "    Jacobian in the origin\t : " << jacobian;
+
+        // If the geometry has valid points, calculate and output its data
+        if (this->AllPointsAreValid()) {
+            Matrix jacobian;
+            this->Jacobian( jacobian, PointType() );
+            rOStream << "    Jacobian in the origin\t : " << jacobian;
+        }
```

Comment:
```suggestion
        // If the geometry has valid points, calculate and output its data
        if (!this->AllPointsAreValid()) {return;}
        Matrix jacobian;
        this->Jacobian( jacobian, PointType() );
        rOStream << "    Jacobian in the origin\t : " << jacobian;
```

I recommend using early exits, helps to reduce the indentation

Reply:
About this, I'd leave it as it is. I think indentation in this case is not a problem as there  is only  a few levels and makes the code more readable IMO.